File: backend/models/booking.model.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

let bookings = new Schema({
    vehiclename: {type: String},
    drivername: {type: String},
    contactno: {type: String},
    priceperh: { type: String},
    hours: {type:String},
    address: {type:String},
    total: {type:String},

  },
   { 
    timestamps: true 
   }
  );
  
  module.exports = mongoose.model("bookings", bookings);

File: agriFront/lib/models/booking.model.dart
class BookingModel {
  late String? id;
  // late String? bNo;
  late String? vehiclename;
  late String? drivername;
  late String? contactno;
  late String? priceperh;
  late String? hours;
  late String? address;
  late String? total;

  BookingModel();
  BookingModel.create(this.id, this.vehiclename, this.drivername,
      this.contactno, this.priceperh, this.hours, this.address, this.total);

  factory BookingModel.fromJson(Map<String, dynamic> json) {
    return BookingModel.create(
        json['_id'],
        json['vehiclename'],
        json['drivername'],
        json['contactno'],
        json['priceperh'],
        json['hours'],
        json['address'],
        json['total']);
  }

  Map<String, dynamic> toJson() => {
        '_id': id,
        'vehiclename': vehiclename,
        'drivername': drivername,
        'contactno': contactno,
        'priceperh': priceperh,
        'hours': hours,
        'address': address,
        'total': total,
      };

  @override
  toString() {
    return vehiclename.toString();
  }
}
